pkg/apis/ceph.rook.io/v1/placement.go

```diff
@@ -25,7 +25,14 @@ func GetMgrPlacement(p rookv1.PlacementSpec) rookv1.Placement {
 }
 
 // GetMonPlacement returns the placement for the MON service
-func GetMonPlacement(p rookv1.PlacementSpec) rookv1.Placement {
+func GetMonPlacement(p rookv1.PlacementSpec, isArbiter bool) rookv1.Placement {
+	// If the mon is the arbiter in a stretch cluster and its placement is specified, return it
+	// without merging with the "all" placement so it can be handled separately from all other daemons
+	if isArbiter {
+		if val, ok := p[KeyMonArbiter]; ok {
+			return val
+		}
+	}
```

Comment:
IMO, the decision to use arbiter or mon placement doesn't quite belong at this code level. I think for the APIs we should stick to things that don't code behavior but are just for utility. I think the behavior of using arbiter vs mon as the placement spec.

Also, I think there is an error in the code as written with it returning `val` instead of returning `p.All().Merge(val)`.

Reply:
> IMO, the decision to use arbiter or mon placement doesn't quite belong at this code level. I think for the APIs we should stick to things that don't code behavior but are just for utility. I think the behavior of using arbiter vs mon as the placement spec.

Sure, the functionality doesn't change, but sounds good for the code organization.

> Also, I think there is an error in the code as written with it returning `val` instead of returning `p.All().Merge(val)`.

The arbiter placement intentionally doesn't merge with the `all`. Arbiter placement is independent. This may not be ideal for some scenarios where you'd have to duplicate some placement, but it will enable other scenarios where you don't want the arbiter to apply placement that applies to all other daemons.